tron/trondaemon.py

```diff
@@ -160,30 +140,36 @@ def __init__(self, options):
         self.manhole_sock = f"{self.options.working_dir}/manhole.sock"
 
     def _build_context(self, options):
-        pidfile = PIDFile(options.pid_file)
         return NoDaemonContext(
+            lockfile=LockFile(options.lock_file),
             working_directory=options.working_dir,
-            umask=0o022,
-            pidfile=pidfile,
             signal_map={
                 signal.SIGHUP: signal.SIG_DFL,
                 signal.SIGINT: signal.default_int_handler,
                 signal.SIGTERM: signal.SIG_DFL,
                 signal.SIGQUIT: signal.SIG_DFL,
                 signal.SIGUSR1: signal.SIG_DFL,
             },
-            files_preserve=[pidfile.lock.file],
         )
 
     def run(self):
         with self.context:
-            setup_logging(self.options)
             self._run_mcp()
             self._run_www_api()
             self._run_manhole()
             self._run_reactor()
 
     def _run_manhole(self):
+        # This condition is made with the assumption that no existing daemon
+        # is running. If there is one, the following code could potentially
+        # cause problems for the other daemon by removing its socket.
+        if (os.path.exists(self.manhole_sock) and
```

Comment:
Hm, this is confusing. What about getting rid of wantPID below, so no lock file is created? We don't really need one anymore. Then you can always delete the manhole_sock file if it exists.

Reply:
Ah, `wantPID=1` is what's causing the lock to be generated. I'll remove it to avoid confusion.